src/problem.cpp

```diff
@@ -293,8 +332,15 @@ std::vector<Solution> Problem::FindSolutionsImpl(Handle& handle,
 
     ret.reserve(found);
 
-    const auto conv_dir = static_cast<conv::Direction>(direction);
-    const auto netcfg   = AsConvolution().BuildConfKey();
+    const auto conv_dir = ([&]() {
+        const auto dir = static_cast<conv::Direction>(direction);
+        if(dir == conv::Direction::BackwardWeights || conv_desc.mode != miopenTranspose)
+            return dir;
+        return dir == conv::Direction::Forward ? conv::Direction::BackwardData
+                                               : conv::Direction::Forward;
+    })();
```

Comment:
I think this logic is also needed inside `AsConvolution()`.

Reply:
Without that, i am still seeing error that it cann't find Invoker. And invoker it is trying to find has `Forward` convolution configuration but invoker was registered as `BackwardData`.

Reply:
I required change similar to this in `AsConvolution` to make this PR work with MIGrpahX. 

https://github.com/ROCmSoftwarePlatform/MIOpen/blob/ef15b6aa7cc29638055f999d52e9ab129f29a8b9/src/problem.cpp#L157